### ISSUE TITLE
change the backoffice theme to my own theme

### ISSUE THREAD

**Issue by ebonit** [PrestaShop/PrestaShop#13020]
**Describe the bug**
when changing the backoffice theme from default to my own theme, it does not show.

I want to use my own theme because I don't like to use template overrides for the backoffice, for instance when adding my custom hooks

**To Reproduce**
1. in the source, go to admin-dev/themes
1. duplicate the backoffice default theme with the name 'mytheme'
2. open the file admin-dev/theme/mytheme/template/controllers/dashboard/helpers/view/view.tpl
3. on line 39 add the line 
`<div class="alert alert-warning">This in mytheme</div>`
4. open the database table ps_employee and change the field bo_theme for the current employee to 'mytheme' (unfortunetly we cannot change themes in the backoffice itself)
5. login into the backoffice as the current employee

**Expected behaviour:** 
on the dashboard we should see an error message 'This is mytheme'

**Real behaviour:**
no error message

**Solution:**
in the file classes/controllers/AdminController.php on line 429 change this line to:
`
$this->bo_theme = ((Validate::isLoadedObject($this->context->employee)
                && $this->context->employee->bo_theme) ? $this->context->employee->bo_theme : $default_theme_name);
`

**Additional information**
PrestaShop version: 7.5.1
PHP version: 7.1.20

**Comment by khouloudbelguith** [comment]
Hi @ebonit,

Thanks for your report.
@PrestaShop/prestashop-core-developers what do you think?

Thanks!

**Comment by khouloudbelguith** [comment]
Hi @ebonit,

I manage to reproduce the issue with PS:1.7.5.1 & PS1.7.4.4.
I will add this issue to our bug roadmap so it will be fixed.
Would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop
Thank you!

**Comment by ebonit** [comment]
I found that the reason why this code was as it is is that all modern controllers use the new-theme templates.

When I use my code as described above, the backoffice breaks on all modern controllers. The only way I can use my own custom backoffice templates is when I set the default value of the 2nd parameter of the constructor to 'mytheme' instead to 'default'
```
 public function __construct($forceControllerName = '', $default_theme_name = 'mytheme')
```
For now I created a override for AdminController where I set the 'default' to 'mytheme'. That works fine. If I find the proper solution then I'll come back with that.

This means that this issue can be closed. my code proposal is not good and for now the override together with the theme setting in ps_employee is the right solution
